src/browser/modules/D3Visualization/lib/visualization/components/graph.js

```diff
@@ -82,7 +82,7 @@ export default class Graph {
         this.nodeMap[eNode.id] = eNode
         this._nodes.push(eNode)
         this.expandedNodeMap[node.id] = this.expandedNodeMap[node.id]
-          ? this.expandedNodeMap[node.id].concat([eNode.id])
+          ? [...new Set(this.expandedNodeMap[node.id].concat([eNode.id]))]
```

Comment:
so I have to investigate a little more but I have removed duplicates at this stage and it seems to work. I have to look into the `nodes` parameter a bit as to why it includes the multiples of the node relationships. It may be that thats where the change is required.

Reply:
I have tracked back to https://github.com/neo4j/neo4j-browser/blob/master/src/shared/services/bolt/boltMappings.js#L214 and tried to put in a `reduce` here to remove duplicates but since this function is used in several places, it seems to cause other issues. So to narrow the scope to the particular problem, Im going to stick with my initial solution for now.

Reply:
That makes sense and I agree with keeping the scope small. 

I think Hugo is tackling it at a lower level in https://github.com/neo4j/neo4j-browser/pull/1003.
So maybe his PR fixes this bug as well @jk05.
Could you have a look if that's the case?

Reply:
Ill take a look 👍

Reply:
I pulled it down locally and the bug still seems to be there...

Reply:
so we can either can look to go ahead with this work to keep the scope smaller, or I can leave a comment on #1003 with a link to this PR (as the context of where Hugo is making the changes could potentially be a better fit than this work). What do you think @oskarhane?

Reply:
I think this change makes sense 👍